src/modules/blockchain/implementation/web3-service.js

```diff
@@ -406,30 +412,42 @@ class Web3Service {
 
     async callContractFunction(contractInstance, functionName, args) {
         let result;
-        while (result === undefined) {
-            try {
-                // eslint-disable-next-line no-await-in-loop
-                result = await contractInstance[functionName](...args);
-            } catch (error) {
-                const decodedErrorData = this._decodeErrorData(error, contractInstance.interface);
-
-                const functionFragment = contractInstance.interface.getFunction(
-                    error.transaction.data.slice(0, 10),
-                );
-                const inputs = functionFragment.inputs
-                    .map((input, i) => {
-                        const argName = input.name;
-                        const argValue = this._formatArgument(args[i]);
-                        return `${argName}=${argValue}`;
-                    })
-                    .join(', ');
+        if (
+            CACHED_FUNCTIONS.ParametersStorage.includes(functionName) &&
+            resultCache[functionName] !== undefined
```

Comment:
Add new method getCachedValue and call it here instead of checking resultCache variable. You can also remove !== undefined